### DISCUSSION TITLE
Is there any way to enforce a MetaMask connector (prevent Coinbase Injected connector override)?

### DISCUSSION THREAD

**Discussion by tylerpashigian** [NoahZinsmeister/web3-react#737] [Q&A] [answered]
I am working under the impression that the Injected connector defaults to MetaMask when installed in the browser, but when using Coinbase Wallet and then going back to the the injected connector, sometimes I get this weird popup from Coinbase Wallet asking if I mean to use them or MetaMask. Is there a MetaMask connector I can use to ensure I am using that wallet rather than relying on Injected? If not, is there a way to prevent Coinbase Wallet from overriding the Injected connector? Screenshot attached using the codepen provided in this repo. Sorry for a potential dumb question, I don't fully understand the Injected connector, any helpful resources would also be appreciated. 
<img width="1437" alt="Screenshot 2023-01-05 at 10 40 08 PM" src="https://user-images.githubusercontent.com/13244107/210931937-e9a5d04a-e33e-4753-98c5-7b6b7e28ace0.png">

**Comment by niZmosis** [comment, marked as answer]
Either way the Injected connector would have to be updated, but there is the metamask/detect-provider which the connector in v8 uses, then Coinbase talks about supporting multiple injected connectors [here](https://docs.cloud.coinbase.com/wallet-sdk/docs/injected-provider-guidance)

**Reply by tylerpashigian** [reply]
Thanks so much, that coinbase link worked for me. It looks like there may be a better way to do it in the uniswap interface front end repo, but I need to use the v6 web3-react package because I need to utilize `MagicConnector` which doesn't appear to be supported in the latest version.
